// @scaleflex/ui/core/switcher-group/switcher-group.component.js
import _extends from "@babel/runtime/helpers/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
var _excluded = ["label", "switcherProps", "readOnly", "disabled"];
import React from 'react';
import PT from 'prop-types';
import { intrinsicComponent } from '../../utils/functions';
import Switcher from '../switcher';
import Styled from './switcher-group.styles';
var SwitcherGroup = intrinsicComponent(function (_ref, ref) {
  var label = _ref.label,
      switcherProps = _ref.switcherProps,
      readOnly = _ref.readOnly,
      disabled = _ref.disabled,
      rest = _objectWithoutProperties(_ref, _excluded);

  return /*#__PURE__*/React.createElement(Styled.SwitcherGroup, {
    ref: ref
  }, /*#__PURE__*/React.createElement(Switcher, _extends({
    switcherProps: switcherProps,
    readOnly: readOnly,
    disabled: disabled
  }, rest)), /*#__PURE__*/React.createElement(Styled.Label, null, label));
});
SwitcherGroup.defaultProps = {
  checked: false
};
SwitcherGroup.propTypes = {
  label: PT.string,
  checked: PT.bool,
  onChange: PT.func,
  switcherProps: PT.object,
  readOnly: PT.bool,
  disabled: PT.bool
};
export default SwitcherGroup;
