// @scaleflex/ui/core/radio/radio.component.js
import _extends from "@babel/runtime/helpers/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
var _excluded = ["checked", "onChange", "radioProps", "readOnly", "disabled"];
import React from 'react';
import PT from 'prop-types';
import RadioButtonIcon from '@scaleflex/icons/radio-button';
import RadioButtonUncheckedIcon from '@scaleflex/icons/radio-button-unchecked';
import { intrinsicComponent } from '../../utils/functions';
import Styled from './radio.styles';
var Radio = intrinsicComponent(function (_ref, ref) {
  var checked = _ref.checked,
      onChange = _ref.onChange,
      radioProps = _ref.radioProps,
      readOnly = _ref.readOnly,
      disabled = _ref.disabled,
      rest = _objectWithoutProperties(_ref, _excluded);

  return /*#__PURE__*/React.createElement(Styled.Radio, _extends({}, rest, {
    ref: ref
  }), /*#__PURE__*/React.createElement(Styled.Input, _extends({
    checked: checked,
    onChange: readOnly || disabled ? undefined : onChange,
    disabled: disabled
  }, radioProps)), checked ? /*#__PURE__*/React.createElement(RadioButtonIcon, null) : /*#__PURE__*/React.createElement(RadioButtonUncheckedIcon, null));
});
Radio.defaultProps = {
  checked: false
};
Radio.propTypes = {
  checked: PT.bool,
  onChange: PT.func,
  // eslint-disable-next-line react/forbid-prop-types
  radioProps: PT.object,
  readOnly: PT.bool,
  disabled: PT.bool
};
export default Radio;
