// @scaleflex/ui/core/switcher/switcher.component.js
import _extends from "@babel/runtime/helpers/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
var _excluded = ["checked", "onChange", "switcherProps", "readOnly", "disabled"];
import React from 'react';
import PT from 'prop-types';
import SwitcherOnIcon from '@scaleflex/icons/switcher-on';
import SwitcherOffIcon from '@scaleflex/icons/switcher-off';
import { intrinsicComponent } from '../../utils/functions';
import Styled from './switcher.styles';
var Switcher = intrinsicComponent(function (_ref, ref) {
  var checked = _ref.checked,
      onChange = _ref.onChange,
      switcherProps = _ref.switcherProps,
      readOnly = _ref.readOnly,
      disabled = _ref.disabled,
      rest = _objectWithoutProperties(_ref, _excluded);

  return /*#__PURE__*/React.createElement(Styled.Switcher, _extends({}, rest, {
    ref: ref
  }), /*#__PURE__*/React.createElement(Styled.Input, _extends({
    checked: checked,
    onChange: readOnly || disabled ? undefined : onChange,
    readOnly: readOnly,
    disabled: disabled
  }, switcherProps)), checked ? /*#__PURE__*/React.createElement(SwitcherOnIcon, null) : /*#__PURE__*/React.createElement(SwitcherOffIcon, null));
});
Switcher.defaultProps = {
  checked: false
};
Switcher.propTypes = {
  checked: PT.bool,
  onChange: PT.func,
  switcherProps: PT.object,
  readOnly: PT.bool,
  disabled: PT.bool
};
export default Switcher;
